src/player.cpp

```diff
@@ -7115,21 +7114,32 @@ void player::hardcoded_effects(effect &it)
             } else if (intense < 1) {
                 it.set_intensity(1);
             }
-            recovery_chance = 24 - intense + 1;
+
+            auto const recovery_chance = 24 - intense + 1;
+
             if (fatigue > 0) {
-                fatigue -= 1 + one_in(recovery_chance);
+                auto const do_roll = [recovery_chance] {
```

Comment:
I don't see wrapping this up in a lambda as making it any less error-prone in the future, and it's certainly more confusing.  I recommend just fixing the code inline, it's only used in two places after all.